plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/PlayAssetDeliveryUtils.java

```
@@ -0,0 +1,73 @@
+ 	limitations under the License.
+*/
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import org.godotengine.godot.Dictionary;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+
+/**
+ * This class contains all the helper methods for serializing/deserializing custom objects used in
+ * the Play Asset Delivery API. The Java objects are serialized into
+ * org.godotengine.godot.Dictionary, which the Godot runtime can receive. For all helper functions,
+ * null is return upon exception.
+ */
+public class PlayAssetDeliveryUtils {
+
+  private static final String TAG = "PlayAssetDeliveryUtils";
+
+  public static Dictionary constructAssetPackStateDictionary(
+      long bytesDownloaded,
+      int errorCode,
+      String name,
+      int status,
+      long totalBytesToDownload,
+      int transferProgressPercentage) {
+    Dictionary returnDict = new Dictionary();
+    returnDict.put("bytesDownloaded", bytesDownloaded);
+    returnDict.put("errorCode", errorCode);
+    returnDict.put("name", name);
+    returnDict.put("status", status);
+    returnDict.put("totalBytesToDownload", totalBytesToDownload);
+    returnDict.put("transferProgressPercentage", transferProgressPercentage);
+    return returnDict;
+  }
+
+  public static Dictionary convertAssetPackStateToDictionary(AssetPackState assetPackState) {
+    return constructAssetPackStateDictionary(
+        assetPackState.bytesDownloaded(),
+        assetPackState.errorCode(),
+        assetPackState.name(),
+        assetPackState.status(),
+        assetPackState.totalBytesToDownload(),
+        assetPackState.transferProgressPercentage());
+  }
+
+  public static AssetPackState convertDictionaryToAssetPackState(Dictionary dict) {
+    try {
+      AssetPackState packState = new AssetPackStateFromDictionary(dict);
+      return packState;
+    } catch (Exception e) {
```

If you do want to return null rather than let the exception bubble up to the caller, I would add a catch block for each exception type you expect, or change AssetPackStateFromDictionary to throw only one type of exception.

---

Now AssetPackStateFromDictionary only throws IllegalArgumentException, and exception here are bubbled up.